mcs/class/System.Security/Test/System.Security.Cryptography.Pkcs/SignedCmsTest.cs

```diff
@@ -374,7 +374,7 @@ public void ComputeEmptySignatureCmsSigner ()
 		}
 
 		[Test]
-		[ExpectedException (typeof (CryptographicException))]
+		[ExpectedException (typeof (PlatformNotSupportedException))]
```

Comment:
Which platforms do we expect this `PlatformNotSupportedException`?

Reply:
All platforms actually. There are two API overloads which require UI interaction. I can rewrite it to use X509Certificate2UI, which is also not implemented and let it throw the exception there.

Reply:
A little more detail for this one. `SignedCms` class has couple of overloads for the `Encrypt` method. The one without parameters is supposed to open an UI for certificate selection and it is not present in CoreFX at all. The same behaviour is also expected in the overload that takes `CmsRecipientCollection` parameter if the collection is empty. Out of convenience I implemented the first overload as `Encrypt(new CmsRecipientCollection())` to maintain the same behaviour for both. `Encrypt(CmsRecipientCollection)` is implemented in CoreFX and throws `PlatformNotSupportedException` for the case that is supposed to open an UI.

Implementing the UI is not really easy to accomplish in a cross-platform manner. I have tried that for Mac/Windows platforms, but eventually I abandoned it because on Mac it relied on the `NSApplication` object being properly initialised beforehand.